Check for __res_query in libresolv before we check for it in libbind.
--- openldap-2.2.23/configure.in	2005-03-01 19:57:10.053436724 -0500
+++ openldap-2.2.23/configure.in	2005-03-01 19:57:27.663137034 -0500
@@ -873,16 +873,6 @@
 fi
 
 if test $ac_cv_func_res_query = no ; then 
-	AC_CHECK_LIB(bind, res_query)
-	ac_cv_func_res_query=$ac_cv_lib_bind_res_query
-fi
-
-if test $ac_cv_func_res_query = no ; then 
-	AC_CHECK_LIB(bind, __res_query)
-	ac_cv_func_res_query=$ac_cv_lib_bind___res_query
-fi
-
-if test $ac_cv_func_res_query = no ; then 
 	AC_CHECK_LIB(resolv, res_query)
 	ac_cv_func_res_query=$ac_cv_lib_resolv_res_query
 fi
@@ -897,6 +887,16 @@
 	ac_cv_func_res_query=$ac_cv_lib_resolv_res_9_query
 fi
 
+if test $ac_cv_func_res_query = no ; then 
+	AC_CHECK_LIB(bind, res_query)
+	ac_cv_func_res_query=$ac_cv_lib_bind_res_query
+fi
+
+if test $ac_cv_func_res_query = no ; then 
+	AC_CHECK_LIB(bind, __res_query)
+	ac_cv_func_res_query=$ac_cv_lib_bind___res_query
+fi
+
 if test "$ac_cv_func_res_query" = yes ; then
 	AC_DEFINE(HAVE_RES_QUERY,1,
 		[define if you have res_query()])
